### DISCUSSION TITLE
Questions about `ThreadDataloader`

### DISCUSSION THREAD

**Discussion by chaoscls** [Project-MONAI/MONAI#6605] [Q&A]
I think `ThreadDataloader` is just use another thread to fetch data, and put it into a buffer queue. I don't know if it really makes sense to do that, or if it is really faster to use `ThreadDataloader`? Because I found there is a similar queue to buffer data in pytorch's Dataloader (https://github.com/pytorch/pytorch/blob/f4228e7037c614ed10b4d5c6fbd1467999e8d3e9/torch/utils/data/dataloader.py#LL1015C4-L1015C4).
So I'm really confused when should I use `ThreadDataloader`?

**Comment by KumoLiu** [comment]
Hi @chaoscls, typical usage is to accelerate light-weight preprocessing (usually cached all the deterministic transforms and no IO operations), because it leverages the separate thread to execute preprocessing to avoid unnecessary IPC between multiple workers of DataLoader. And as CUDA may not work well with the multi-processing of DataLoader, `ThreadDataLoader` can be useful for GPU transforms. 
For more details, you can refer to https://github.com/Project-MONAI/tutorials/blob/main/acceleration/fast_model_training_guide.md#4-threaddataloader-vs-dataloader.

Hope it can help you, thanks!
